### PR TITLE
Feature - First MQTT Draft

### PR DESCRIPTION
**Pull request by pLeminoq**
### :scroll: Description

As the title says this PR replaces all RSB classes and interfaces with MQTT.

### :hammer: Breaking Changes
* All RSB dependencies and classes were removed.
* RPCMethod Annotation no longer works on methods of interfaces with generics (e.q. `M requestStatus()`)

### CODE REVIEW

File: communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractControllerServer.java [hunk 1]
```diff
@@ -178,69 +185,71 @@ public void init(final String scope) throws InitializationException, Interrupted
...
+                server.registerMethods(Pingable.class, this);
             } catch (InvalidStateException ex) {
                 // if already registered then everything is fine and we can continue...
             }
             try {
-                RPCHelper.registerInterface(Requestable.class, this, server);
-            } catch (InvalidStateException ex) {
+                server.registerMethods((Class) getClass(), this);
+                //TODO: this does not work since the requestable interface has a generic return type
+                //server.registerMethods(Requestable.class, this);
```
Comment (DivineThreepwood) [inline communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractControllerServer.java]: Cleanup code or create and ref issue

File: communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractControllerServer.java [hunk 2]
```diff
@@ -1238,10 +1247,9 @@ public boolean isShutdownInProgress() {
...
     @Override
     public String toString() {
-        try {
-            return getClass().getSimpleName() + "[" + informer.getScope().toString() + "]";
-        } catch (NotAvailableException ex) {
-            return getClass().getSimpleName() + "[]";
+        if (publisher == null) {
+            return getClass().getSimpleName();
         }
+        return getClass().getSimpleName() + "[" + publisher.getScope() + "]";
```
Comment (pLeminoq) [inline communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractControllerServer.java:1249]: I think `publisher.getScope()` now returns our scope proto type whose toString() prints a multi line string. Maybe we this should use the `ScopeProcesser.generateStringRep()` method.

File: communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractRemoteClient.java [hunk 3]
```diff
@@ -266,24 +263,24 @@ private void internalInit(final Scope scope, final ParticipantConfig participant
...
+    private void initRemoteServer(final Scope scope, final CommunicatorConfig communicatorConfig) throws CouldNotPerformException {
         try {
-            this.remoteServer = RSBFactoryImpl.getInstance().createSynchronizedRemoteServer(scope.concat(AbstractControllerServer.SCOPE_SUFFIX_CONTROL), participantConfig);
-            this.remoteServerWatchDog = new WatchDog(remoteServer, "RSBRemoteServer[" + scope.concat(AbstractControllerServer.SCOPE_SUFFIX_CONTROL) + "]");
-            this.listenerWatchDog.addObserver(middlewareReadyObserver);
-            this.listenerWatchDog.addObserver(middlewareFailureObserver);
-            this.remoteServerWatchDog.addObserver(middlewareFailureObserver);
-        } catch (RuntimeException | InstantiationException ex) {
-            throw new CouldNotPerformException("Could not create RemoteServer on scope [" + scope + "]!", ex);
+            this.rpcClient = factory.createRPCClient(ScopeProcessor.concat(scope, AbstractControllerServer.SCOPE_SUFFIX_CONTROL), communicatorConfig);
```
Comment (pLeminoq) [inline communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractRemoteClient.java:277]: We could remove the scope suffix as the MQTT RPC implementation add `/rpc` to its initialization scope anyway.

File: communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractRemoteClient.java [hunk 4]
```diff
@@ -266,24 +263,24 @@ private void internalInit(final Scope scope, final ParticipantConfig participant
...
+            this.rpcClientWatchDog = new WatchDog(rpcClient, "RPCClient[" + ScopeProcessor.generateStringRep(rpcClient.getScope()) + "]");
+            this.subscriberWatchDog.addObserver(middlewareReadyObserver);
+            this.subscriberWatchDog.addObserver(middlewareFailureObserver);
+            this.rpcClientWatchDog.addObserver(middlewareFailureObserver);
+        } catch (RuntimeException ex) {
+            throw new CouldNotPerformException("Could not create RPCClient on scope [" + scope + "]!", ex);
```
Comment (pLeminoq) [inline communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractRemoteClient.java:283]: Scope should be replaced by string rep of `ScopeProcessor`

**Review by DivineThreepwood — APPROVED** [review]

**Review by pLeminoq — COMMENTED** [review]
Looks good to me.

**Pull request merged**